### DISCUSSION TITLE
Skia.Gtk head "cannot open display" error

### DISCUSSION THREAD

**Discussion by igiona** [unoplatform/uno#12235] [Q&A] [answered]
### Current behavior

When I try run&debug (via WSL) my GTK App I get the following error and message in the output window in VisualStudio:

`(MyBRO.Skia.Gtk:101): Gtk-WARNING **: 05:03:36.552: cannot open display: `
![image](https://github.com/unoplatform/uno/assets/1524038/49654c48-5a30-4080-bb18-62ab6b45c428)

I can be reproduce on a empty app based on the Uno template.

NOTE: This used to work, I'm not sure if any change in WSL/Windows/VisualStudio has affected this.

### Expected behavior

The GTK app starts and can be debugged within VisualStudio

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report. This generally means that you either do not have an X11 server running (see [this document](https://platform.uno/docs/articles/get-started-with-linux.html?tabs=ubuntu1804#setting-for-windows-and-wsl)), or that you do you not have [WSLg setup properly](https://github.com/microsoft/wslg#installing-wslg).

**Reply by igiona** [reply]
@jeromelaban thanks for you support.

I managed to make the GTK app work again, these are the steps I went through, unfortunately I cannot say which one really did it for me:

As explained here: https://github.com/microsoft/wslg

I changed my installation to WSL2
`wsl --set-version "Ubuntu-22.04" 2`

Reboot&Update WSL
```
wsl --shutdown
wsl --update
```

EDIT: After a bit of testing, I can now assert that (at least for me) WSL must be set to V2 for the GTK Head to work.
